File: asyncSrc/passwordReset/logic.js
const { hash } = require('/opt/hashing');
const {
  createResetToken,
  readApplication,
  sendResetPasswordEmail,
} = require('/opt/ports');

/**
 *
 * @param {string} applicationId Application ID
 * @param {string} email Email address to send reset password token
 * @returns
 */
async function logic(applicationId, email) {
  const emailHash = hash(email);
  const resetToken = await createResetToken(applicationId, emailHash);

  const applicationData = await readApplication(applicationId);
  const {
    resetPasswordUrl,
  } = applicationData;
  await sendResetPasswordEmail(email, resetToken, resetPasswordUrl);
}

module.exports = {
  logic,
};

File: src/cmd/requestPasswordReset/logic.go
package main

import (
	"strings"

	"github.com/thomasstep/authentication-service/internal/adapters"
)

func logic(applicationId string, email string) {
	lowerCaseEmail := strings.ToLower(email)
	// create reset token record
	resetToken, createRecordErr := adapters.CreateResetPasswordRecord(applicationId, lowerCaseEmail)
	if createRecordErr != nil {
		panic(createRecordErr)
	}
	// read application for the reset password url
	applicationItem, readAppErr := adapters.ReadApplication(applicationId)
	if readAppErr != nil {
		panic(readAppErr)
	}
	// send reset password email
	adapters.SendResetPasswordEmail(lowerCaseEmail, resetToken, applicationItem.ResetPasswordUrl)
}
